sycl/include/sycl/platform.hpp

```diff
@@ -46,15 +53,30 @@ class __SYCL_EXPORT platform {
   explicit platform(cl_platform_id PlatformId);
 #endif
 
-  /// Constructs a SYCL platform instance using device selector.
+  /// Constructs a SYCL platform instance using a device_selector.
   ///
   /// One of the SYCL devices that is associated with the constructed SYCL
   /// platform instance must be the SYCL device that is produced from the
   /// provided device selector.
   ///
-  /// \param DeviceSelector is an instance of SYCL device_selector.
+  /// \param DeviceSelector is an instance of a SYCL 1.2.1 device_selector
   explicit platform(const device_selector &DeviceSelector);
 
+#if __cplusplus >= 201703L
+  /// Constructs a SYCL platform instance using the platform of the device
+  /// identified by the device selector provided.
+  /// \param DeviceSelector is SYCL 2020 Device Selector, a simple callable that
+  /// takes a device and returns an int
+  template <
+      typename DeviceSelector,
+      typename = std::enable_if_t<
+          std::is_invocable_r<int, DeviceSelector &, device &>::value &&
+          !std::is_base_of_v<ext::oneapi::filter_selector, DeviceSelector>>>
+  // see [filter_selectors not "Callable"] in device_selectors.cpp
+  explicit platform(const DeviceSelector &deviceSelector)
+      : platform(detail::select_device(deviceSelector)) {}
```

Comment:
Why this does not work?
```suggestion
      : platform(device { deviceSelector }) {}
```

Reply:
The honest answer is I didn't think of it.  But device is an incomplete type here in platform.hpp, so that's the reason it won't work.